### PR TITLE
App Integrity e2e test cases

### PR DESCRIPTION
**Pull request by spetrov**
# JIRA Ticket

[SDKS-2785](https://bugster.forgerock.org/jira/browse/SDKS-2785) App Integrity QA

# Description

Implemented e2e test cases covering the App Integrity feature. The test cases require "live" AM server with the following tree: 

image

### CODE REVIEW

File: FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA_07_AppIntegrityTest.swift [hunk 1]
```diff
@@ -177,7 +921,7 @@ final class AA_07_AppIntegrityTest: CallbackBaseTest {
         // Provide input value for the username collector callback
         for callback in node.callbacks {
             if callback is NameCallback, let nameCallback = callback as? NameCallback {
-                nameCallback.setValue(AA_07_AppIntegrityTest.USERNAME)
+                nameCallback.setValue(AA_05_DeviceBindingCallbackTest.USERNAME)
```
Comment (vahancouver) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA_07_AppIntegrityTest.swift:924]: should this be `AA_07_AppIntegrityTest.USERNAME`?

**Review by vahancouver — COMMENTED** [review]
small comment

File: FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA_07_AppIntegrityTest.swift (hunk 1 above)
Comment (spetrov) [reply FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA_07_AppIntegrityTest.swift]: Fixed.

**Review by vahancouver — APPROVED** [review]
LGTM

File: FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA-05-DeviceBindingCallbackTest.swift [hunk 2]
```diff
@@ -71,7 +71,7 @@ class AA_05_DeviceBindingCallbackTest: CallbackBaseTest {
         for callback in currentNode.callbacks {
             if callback is DeviceBindingCallback, let deviceBindingCallback = callback as? DeviceBindingCallback {
                 XCTAssertNotNil(deviceBindingCallback.userId)
-                XCTAssertEqual(deviceBindingCallback.userName, AA_05_DeviceBindingCallbackTest.USERNAME)
+//                XCTAssertEqual(deviceBindingCallback.userName, AA_05_DeviceBindingCallbackTest.USERNAME)
```
Comment (jeyanthanperiyasamy) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA-05-DeviceBindingCallbackTest.swift:74]: should we remove this ?

File: FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA-05-DeviceBindingCallbackTest.swift [hunk 3]
```diff
@@ -117,7 +117,7 @@ class AA_05_DeviceBindingCallbackTest: CallbackBaseTest {
             if callback is DeviceBindingCallback, let deviceBindingCallback = callback as? DeviceBindingCallback {
                 
                 XCTAssertNotNil(deviceBindingCallback.userId)
-                XCTAssertEqual(deviceBindingCallback.userName, AA_05_DeviceBindingCallbackTest.USERNAME)
+//                XCTAssertEqual(deviceBindingCallback.userName, AA_05_DeviceBindingCallbackTest.USERNAME)
```
Comment (jeyanthanperiyasamy) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA-05-DeviceBindingCallbackTest.swift:120]: remove this ?

**Review by jeyanthanperiyasamy — COMMENTED** [review]
many scenarios covered ..nice job .

**Review by jeyanthanperiyasamy — APPROVED** [review]

File: FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA-05-DeviceBindingCallbackTest.swift (hunk 3 above)
Comment (spetrov) [reply FRAuth/FRAuthTests/FRAuthSwiftTests/E2ETests/Callback-Live/AA-05-DeviceBindingCallbackTest.swift:120]: Same as above...

**Pull request merged**